From 0bb30410cc882dab7836d18baa5b987bcf63d96b Mon Sep 17 00:00:00 2001
From: Andrew Low <contact@ackl.me>
Date: Fri, 12 May 2017 16:23:25 +0100
Subject: [PATCH] vala 0.36 requires ref keyword in function param (#33)

* vala 0.36 requires ref keyword in function param

* wrap with if clauses for maximum support
---
 src/font-manager/Compare.vala | 8 ++++++++
 1 file changed, 8 insertions(+)

diff --git a/src/font-manager/Compare.vala b/src/font-manager/Compare.vala
index 841243e..9246d18 100755
--- a/src/font-manager/Compare.vala
+++ b/src/font-manager/Compare.vala
@@ -280,9 +280,17 @@ namespace FontManager {
                  * the iter was always being set to null after calling remove.
                  */
                 string iter_as_string = store.get_string_from_iter(iter);
+#if VALA_0_36
+                store.remove(ref iter);
+#else
                 store.remove(iter);
+#endif
                 store.get_iter_from_string(out iter, iter_as_string);
+#if VALA_0_36
+                bool still_valid = store.remove(ref iter);
+#else
                 bool still_valid = store.remove(iter);
+#endif
                 /* Set the cursor to a remaining row instead of having the cursor disappear.
                  * This allows for easy deletion of multiple previews by hitting the remove
                  * button repeatedly.
